compiler/noirc_evaluator/src/ssa/checks/check_for_underconstrained_values.rs

```diff
@@ -354,83 +732,268 @@ impl Context {
-        //  }
-        //
-        //  fn main {
-        //   b0(v0: Field, v1: Field):
-        //      v2 = add v0, 1
-        //      v3 = mul v1, 2
-        //      v4 = call br(v2, v3)
-        //      v5 = add v4, 2
-        //      return
-        // }
-        let main_id = Id::test_new(0);
-        let mut builder = FunctionBuilder::new("main".into(), main_id);
-        let v0 = builder.add_parameter(Type::field());
-        let v1 = builder.add_parameter(Type::field());
-
-        let one = builder.field_constant(1u128);
-        let two = builder.field_constant(2u128);
-
-        let v2 = builder.insert_binary(v0, BinaryOp::Add, one);
-        let v3 = builder.insert_binary(v1, BinaryOp::Mul, two);
-
-        let br_function_id = Id::test_new(1);
-        let br_function = builder.import_function(br_function_id);
-        let v4 = builder.insert_call(br_function, vec![v2, v3], vec![Type::field()])[0];
-        let v5 = builder.insert_binary(v4, BinaryOp::Add, two);
-        builder.insert_constrain(v5, one, None);
-        builder.terminate_with_return(vec![]);
-
-        builder.new_brillig_function("br".into(), br_function_id, InlineType::default());
-        let v0 = builder.add_parameter(Type::field());
-        let v1 = builder.add_parameter(Type::field());
-        let v2 = builder.insert_binary(v0, BinaryOp::Add, v1);
-        builder.terminate_with_return(vec![v2]);
-        let mut ssa = builder.finish();
+        let program = r#"
+        acir(inline) fn main f0 {
+            b0(v0: Field, v1: Field):
+                v2 = add v0, Field 1
+                v3 = mul v1, Field 2
+                v4 = call f1(v2, v3) -> Field
+                v5 = add v4, Field 2
+                return
+        }
+        
+        brillig(inline) fn br f1 {
+          b0(v0: Field, v1: Field):
+            v2 = add v0, v1
+            return v2
+        }
+        "#;
+
+        let mut ssa = Ssa::from_str(program).unwrap();
         let ssa_level_warnings = ssa.check_for_underconstrained_values();
         assert_eq!(ssa_level_warnings.len(), 1);
     }
+
+    #[test]
+    #[traced_test]
+    /// Test where a call to a Brillig function is left unchecked with a later assert,
+    /// by example of the program illustrating issue #5425 (simplified variant).
+    fn test_underconstrained_value_detector_5425() {
+        /*
+        unconstrained fn maximum_price(options: [u32; 2]) -> u32 {
+            let mut maximum_option = options[0];
+            if (options[1] > options[0]) {
+                maximum_option = options[1];
+            }
+            maximum_option
+        }
+
+        fn main(sandwiches: pub [u32; 2], drinks: pub [u32; 2], best_value: u32) {
+            let most_expensive_sandwich = maximum_price(sandwiches);
+            let mut sandwich_exists = false;
+            sandwich_exists |= (sandwiches[0] == most_expensive_sandwich);
+            sandwich_exists |= (sandwiches[1] == most_expensive_sandwich);
```

Comment:
Is there a test that shows what happens if only 1 element of the input array is constrained?

Reply:
No, checking individual array elements isn't supported currently. It's been moved into a separate issue: https://github.com/noir-lang/noir/issues/6698